Apps/phcorelight/corelight_connector.py

```diff
@@ -0,0 +1,573 @@
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        append = param['append']
+        name = param['name']
+        input = param['input']
+
+        if (append):
+            f = open("{0}/{1}".format(self._local_dir, name), "w")
+            f.write(input)
+            files = {'file': f}
+            ret_val, response = self._make_rest_call('/api/bro/inpu' + name, action_result, params=None, headers=None, method="put", files=files)
+        else:
+            f = open("{0}/{1}".format(self._local_dir, name), "w")
+            fields = param.get('fields')
+            f.write(fields + "\n")
+            f.write(input)
+            files = {'file': f}
+            ret_val, response = self._make_rest_call('/api/bro/inpu' + name, action_result, params=None, headers=None, method="post", files=files)
+
+        f.close()
+
+        if (phantom.is_fail(ret_val)):
+            return action_result.get_status()
+
+        action_result.add_data(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_backup_corelight(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        bundlePassword = param['bundle_password']
+        uri = '/api/provision/bundle?bundle-password=' + bundlePassword + '&no-sensitive=0&type=backup'
+        ret_val, response = self._make_rest_call(uri, action_result, params=None, headers=None)
+
+        try:
+            fileName = param['backup_name']
+            file_path = "{0}/{1}".format(self._local_dir, fileName)
+            output = json.dumps(response)
+            with open(file_path, 'wb') as f:
+                f.write(output)
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, "Unable to write to a temporary file in the folder {0}".format(self._local_dir), e)
+
+        vault_ret_dict = Vault.add_attachment(file_path, self.get_container_id(), file_name=fileName)
+        curr_data = {}
+        if (vault_ret_dict['succeeded']):
+            curr_data[phantom.APP_JSON_VAULT_ID] = vault_ret_dict[phantom.APP_JSON_HASH]
+            curr_data[phantom.APP_JSON_NAME] = fileName
+            action_result.add_data(curr_data)
+            wanted_keys = [phantom.APP_JSON_VAULT_ID, phantom.APP_JSON_NAME]
+            summary = {x: curr_data[x] for x in wanted_keys}
+            action_result.update_summary(summary)
+            action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            action_result.set_status(phantom.APP_ERROR, phantom.APP_ERR_FILE_ADD_TO_VAULT)
+            action_result.append_to_message(vault_ret_dict['message'])
+
+        # remove the /tmp/<> temporary directory
+        shutil.rmtree(self._local_dir)
```

Comment:
This needs to be moved to finalize to make sure we clean up after action runs

Reply:
Moved